trivup/apps/OauthbearerOIDCApp.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python
+#
+
+# Copyright (c) 2021, Magnus Edenhill
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from trivup import trivup
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import jwt
+import datetime
+import json
+import argparse
+import requests
+from Crypto.PublicKey import RSA
+
+public_keys = []
+
+
+class WebServerHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text')
+        self.end_headers()
+        message = "HTTP server for OAuth\n"
+        message += "Example for token retrieval:\n"
+        message += 'curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "Accept: application/json" \
+        -H "Authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "Cache-Control: no-cache" \
+        -d "method=oidc,scope=test-scope"'
+        self.wfile.write(message.encode())
+        print(message)
+
+    def generate_valid_token_for_client(self):
+        """
+        Example usage:
+        curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "Accept: application/json" \
+        -H "Authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "Cache-Control: no-cache" \
+        -d "method=oidc,scope=test-scope"
+        """
+        if self.headers.get('Content-Length', None) is None:
+            self.send_error(404, 'Content-Length field is required')
```

Comment:
One could pedantically argue this should be a `400` status code, but since it's for testing, it probably doesn't matter.

The same argument applies for the other places that `404` is being returned.